app/gui2/src/util/callTree.ts

```diff
@@ -395,6 +396,53 @@ export function getAccessOprSubject(app: Ast.Ast): Ast.Ast | undefined {
   if (app instanceof Ast.PropertyAccess) return app.lhs
 }
 
+/** Same as {@link GraphDb.getMethodCallInfo} but with a special handling for nested Applications.
+ * Sometimes we receive MethodCallInfo for inner sub-applications of the expression,
+ * and we want to reuse it for outer application expressions when adding new arguments to the call.
+ * It requires adjusting `notAppliedArguments` array, but otherwise is a straightforward recursive call.
+ * We stop recursion at any not-application AST. We expect that a subexpression’s info is only valid if it is a part of the prefix application chain.
+ * We also don’t consider infix applications here, as using them inside a prefix chain would require additional syntax (like parenthesis). */
+export function getMethodCallInfoRecursively(
+  ast: Ast.Ast,
+  db: GraphDb,
+): MethodCallInfo | undefined {
+  const get = (
+    ast: Ast.Ast,
+    appliedArgs: number,
+    appliedNamedArgs: string[],
+  ): MethodCallInfo | undefined => {
+    const info = db.getMethodCallInfo(ast.id)
+    if (info) {
+      // There is an info available! Stop the recursion and adjust `notAppliedArguments`.
+      // Indices of all named arguments applied so far.
+      const appliedNamed =
+        appliedNamedArgs.length > 0 ?
+          info.suggestion.arguments
+            .map((arg, index) => (appliedNamedArgs.includes(arg.name) ? index : -1))
+            .filter((i) => i !== -1)
+        : []
+      const correctedNotAppliedArguments = info.methodCall.notAppliedArguments.filter(
+        (idx) => !appliedNamed.includes(idx),
+      )
+      return {
+        methodCall: {
+          ...info.methodCall,
+          notAppliedArguments: correctedNotAppliedArguments.sort().slice(appliedArgs),
+        },
+        suggestion: info.suggestion,
+        partiallyApplied: info.partiallyApplied,
```

Comment:
On self-review I noticed that I’m not doing anything with this field. The issue is that I don’t quite understand its purposes and how it should be changed for outer expressions when it’s coming from inner ones.